vizro-core/src/vizro/models/_navigation/navigation.py

```diff
@@ -37,11 +38,12 @@ def pre_build(self):
 
     @_log_call
     def build(self, *, active_page_id=None) -> _NavBuildType:
-        nav_selector = self.nav_selector.build(active_page_id=active_page_id)
+        nav_selector: NavSelectorType = cast(NavSelectorType, self.nav_selector).build(active_page_id=active_page_id)
```

Comment:
This doesn't look right: `nav_selector` here is the _built_ version, i.e. a dash component and not itself a `NavSelectorType`. Is this what you meant?
```suggestion
        nav_selector = cast(NavSelectorType, self.nav_selector).build(active_page_id=active_page_id)
```

Reply:
P.S. I think maybe one of my reasons for removing the `Optional` was so that we didn't need to do this `cast` here at all.

Reply:
Ah yes! We do something very similar in `_dashboard.py`:

`navigation: _NavBuildType = cast(Navigation, self.navigation).build(active_page_id=page.id)`

I am also fine with removing the `Optional`, but then we should do it through the entire code basis because in the Dashboard model we currently also have this:

`navigation: Optional[Navigation] = None`

So either we remove it everywhere or we keep it?

Reply:
Followed up here: https://github.com/mckinsey/vizro/pull/190#discussion_r1416628062